### PR TITLE
Fix bug with Solaris

### PR DESCRIPTION
**Pull request by JonathanMeans**

### CODE REVIEW

File: six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp [hunk 1]
```diff
@@ -129,14 +129,16 @@ class Tester
...
-        mData(createData<DataTypeT>(mDims)),
         mImage(mDims.area()),
         mTestPathname("streaming_write.nitf"),
         mSchemaPaths(schemaPaths),
         mSetMaxProductSize(setMaxProductSize),
         mMaxProductSize(maxProductSize),
         mSuccess(true)
     {
+        // This can't go in the initializer list, because Solaris complains
+        mData = createData<DataTypeT>(mDims);
```
Comment (asylvest) [inline six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp]: Is this a compile time or runtime error on Solaris..?

Comment (JonathanMeans) [reply six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp]: Runtime. It says "Illegal instruction" and aborts.

Comment (asylvest) [reply six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp]: That is really weird (and philosophically annoying).  If you leave it on the initializer list but call `createData.release()`, does that work?

Comment (JonathanMeans) [reply six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp]: Yes!

Comment (JonathanMeans) [reply six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp]: [This](https://docs.oracle.com/cd/E19205-01/820-4155/c++_faq.html#LibComp3) may be related?

Comment (asylvest) [reply six/modules/c++/six.sicd/tests/test_sicd_byte_provider.cpp]: OK glad we've got a workaround that is at least just slightly off from what we'd normally do.  Hmm, yeah you're probably right that there's some piece of the standard they're not implementing for... whatever reason.

**Comment by JonathanMeans** [comment]
Comment added.

**Comment by asylvest** [comment]
Cool - thanks.

**Pull request merged**